inference-engine/tests/functional/plugin/shared/src/single_layer_tests/psroi_pooling.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tuple>
+#include <string>
+#include <vector>
+#include <memory>
+
+#include "common_test_utils/common_utils.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+#include "functional_test_utils/layer_test_utils.hpp"
+
+#include "single_layer_tests/psroi_pooling.hpp"
+
+using namespace InferenceEngine;
+using namespace FuncTestUtils::PrecisionUtils;
+
+namespace LayerTestsDefinitions {
+
+    std::string PSROIPoolingLayerTest::getTestCaseName(testing::TestParamInfo<psroiParams> obj) {
+        std::vector<size_t> inputShape;
+        std::vector<size_t> coordsShape;
+        size_t outputDim;
+        size_t groupSize;
+        float spatialScale;
+        size_t spatialBinsX;
+        size_t spatialBinsY;
+        std::string mode;
+        InferenceEngine::Precision netPrecision;
+        std::string targetDevice;
+        std::tie(inputShape, coordsShape, outputDim, groupSize, spatialScale, spatialBinsX, spatialBinsY, mode, netPrecision, targetDevice) = obj.param;
+
+        std::ostringstream result;
+
+        result << "in_shape=" << CommonTestUtils::vec2str(inputShape) << "_";
+        result << "coord_shape=" << CommonTestUtils::vec2str(coordsShape) << "_";
+        result << "out_dim=" << outputDim << "_";
+        result << "group_size=" << groupSize << "_";
+        result << "scale=" << spatialScale << "_";
+        result << "bins_x=" << spatialBinsX << "_";
+        result << "bins_y=" << spatialBinsY << "_";
+        result << "mode=" << mode << "_";
+        result << "prec=" << netPrecision.name() << "_";
+        result << "dev=" << targetDevice;
+        return result.str();
+    }
+
+    static int randInt(int low, int high) {
+        std::random_device rd;
+        std::mt19937 gen(rd());
+        std::uniform_int_distribution<int> dis(low, high);
+        return dis(gen);
+    }
+
+    static void fillROITensor(float* buffer, int numROIs, int batchSize,
```

Comment:
I saw the same function in [ROIPooling SLT](https://github.com/openvinotoolkit/openvino/pull/2903/files#diff-ca3ee0b2ce05653ecf551a1fe95161e788d7c9582447bda4664bdd13629b9639). Could you reuse it?

Reply:
I'll use it once that PR gets merged

Reply:
So actually I cannot use function from ROIPooling SLT since ROIs for PSROIPooling depend on different params (like scale, mode, spatial bins)